sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -3141,3 +3141,88 @@ case class TimestampAdd(
     copy(unit = newFirst, quantity = newSecond, timestamp = newThird)
   }
 }
+
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "_FUNC_(unit, startTimestamp, endTimestamp) - Gets the difference between the timestamps `endTimestamp` and `startTimestamp` in the specified units.",
+  arguments = """
+    Arguments:
+      * unit - this indicates the units of the difference between the given timestamps.
```

Comment:
shall we define the rounding behavior? e.g. if the actual result is 1.6 year, what do we return?

Reply:
The behavior has been described in PR's description already:
"... The function calculates the amount in terms of this unit, and returns a **whole** number, representing the number of **complete units between the two timestamps**. "

Reply:
end-users can't see PR descriptions... We need to put them in the function doc.

BTW what does "whole number" mean? 1.6 years will become 2 years?

Reply:
whole and complete means 1 year for 1.6 years.

Reply:
ok so it simply truncates the fraction part. Can we clarify it in the function doc?